src/wast-parser.cc

```diff
@@ -874,7 +874,7 @@ bool WastParser::ParseElemExprOpt(ExprList* out_elem_expr) {
     }
     EXPECT(Rpar);
   } else {
-    if (ParseExpr(&exprs) != Result::Ok) {
+    if (IsExpr(PeekPair()) && ParseExpr(&exprs) != Result::Ok) {
```

Comment:
Should this maybe be `if (!IsExpr(PeekPair()) ||`?

Could you maybe add a test case and update the PR description and title to explain a little bit what this is fixing?